funnel/views/session.py

```diff
@@ -17,15 +17,19 @@ def rooms_list(project):
 
 
 def session_form(project, proposal=None, session=None):
-    if session:
+    # Look for any existing unscheduled session
+    if proposal and not session:
+        session = Session.query.filter_by(project=project, proposal=proposal).first()
```

Comment:
Add a classmethod to Session and make this call `Session.for_proposal(proposal, create=True)`